### PR TITLE
fix: use taplo for formatting Cargo.toml

### PR DESCRIPTION
**Pull request by Bravo555**
## Proposed changes

cargo-sort was replaced by [taplo](https://taplo.tamasfe.dev/), which besides sorting dependencies, can also perform TOML reformatting.

New rules for formatting Cargo.toml files are:

- indent 4 spaces
- only split into multiple lines if content doesn't fit in a single line
- reorder only dependencies
## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/common/tedge_config_macros/macro/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ repository = { workspace = true }
 proc-macro = true
 
 [dependencies]
-tedge_config_macros-impl = { workspace = true }
-syn = { workspace = true }
-quote = { workspace = true }
 proc-macro2 = { workspace = true }
+quote = { workspace = true }
```
Comment (albinsuresh) [inline crates/common/tedge_config_macros/macro/Cargo.toml:16]: Looks like the sorting logic of `cargo-sort` was really broken, even for obvious things like this. Glad we replaced it with something more sensible.

File: justfile [hunk 2]
```diff
@@ -25,7 +25,7 @@ default:
 # Install necessary tools
 install-tools:
     rustup component add rustfmt --toolchain nightly
-    cargo install cargo-sort cargo-nextest
+    cargo install taplo-cli cargo-nextest
```
Comment (albinsuresh) [inline justfile:28]: The following recipes are still using `cargo sort` command: `check-tools`, `format` and `format-check`.They must be replaced with `taplo fmt --check` instead, right?

**Review by albinsuresh — CHANGES_REQUESTED** [review]

**Review by albinsuresh — APPROVED** [review]

**Review by didier-wenzek — APPROVED** [review]
Approved. Thank you! Running `cargo sort` was really useless :-(

**Review by jarhodes314 — APPROVED** [review]
Nice to see this gives us actual TOML formatting too, not just sorting (as well as sorting in more places than we had)!

**Pull request merged**